.release-notes/4167.md

```diff
@@ -0,0 +1,22 @@
+## Fix for compiler segfault #4130
```

Comment:
I'd omit the PR number from the title (you can keep it in the next paragraph) and try to summarize what is being fixed, along the lines of "Fix compiler crash related to using private types as default arguments".